### PR TITLE
Adds hauberoaches, cockroaches wearing spiked helmets

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Sure, glockroaches are cool and all, but that gun doesn't do you much good once someone manages to close the distance on you. Thus, meet the newest member of the cockroach family tree, a veritable tank, the hauberoach! Careful, they're bruisers!

See them in action here:
https://cdn.discordapp.com/attachments/585862469508005888/731427237626249246/hauberoach.mp4

Squishing these boys by stepping on them requires having the PIERCEIMMUNE flag, those are high tech monomolecular spikes on those helmets! They didn't do that with the intention of piercing shoes and boots, it's just a result of how tiny these helmets are. Otherwise, you get caltrop'd and knocked over defenseless, and the hauberoach can gore you to death or lay its eggs in your mouth or whatever cockroaches do when they maim people. You can also probably just smack them I think.

## Why It's Good For The Game
Expanded cockroach gameplay and class types for deeper replayability value/admin events

## Changelog
:cl: Ryll/Shaps
add: Be on the lookout for new types of cockroaches wearing miniature 19th century Prussian headgear!
/:cl:

### CODE REVIEW

**Comment by zeskorion** [comment]
having pierceimmune will only make you do damage to the roach, it won't stop the roach's caltrop

**Comment by Ryll-Ryll** [comment]
The caltrop component has its own PIERCEIMMUNE check that prevents the damage (though it can still attack you manually)

File: code/modules/mob/living/simple_animal/hostile/cockroach.dm [hunk 1]
```diff
@@ -63,25 +63,54 @@
...
+/mob/living/simple_animal/hostile/cockroach/hauberoach
+	name = "hauberoach"
+	desc = "Is that cockroach wearing a tiny yet immaculate replica 19th century Prussian spiked helmet? ...Is that a bad thing?"
+	icon_state = "hauberoach"
+	attack_verb_continuous = "rams its spike into"
+	attack_verb_simple = "ram your spike into"
+	melee_damage_lower = 5
+	melee_damage_upper = 20
+	obj_damage = 20
+	gold_core_spawnable = HOSTILE_SPAWN
```
Comment (ATH1909) [inline code/modules/mob/living/simple_animal/hostile/cockroach.dm:96]: based

File: code/modules/mob/living/simple_animal/hostile/cockroach.dm [hunk 2]
```diff
@@ -63,25 +63,54 @@
...
+/mob/living/simple_animal/hostile/cockroach/hauberoach/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/caltrop, 10, 15, 100, (CALTROP_BYPASS_SHOES | CALTROP_SILENT))
+
+/mob/living/simple_animal/hostile/cockroach/hauberoach/Crossed(var/atom/movable/AM)
+	var/mob/living/A = AM
+	if(istype(A) && A.mob_size > MOB_SIZE_SMALL && !(A.movement_type & FLYING))
+		if(!HAS_TRAIT(A, TRAIT_PIERCEIMMUNE))
+			A.visible_message("<span class='danger'>[A] steps onto [src]'s spike!</span>", "<span class='userdanger'>You step onto [src]'s spike!</span>")
+		else if(HAS_TRAIT(A, TRAIT_PACIFISM))
```
Comment (ATH1909) [inline code/modules/mob/living/simple_animal/hostile/cockroach.dm:111]: imo, you should swap the position of this with the position of the pierceimmune check, if possible

Comment (Ryll-Ryll) [reply code/modules/mob/living/simple_animal/hostile/cockroach.dm:111]: since when did the prussians honor peace treaties?

Comment (ATH1909) [reply code/modules/mob/living/simple_animal/hostile/cockroach.dm:111]: it's less that and more that if a pacifist is intentionally trying to NOT step on the hauberoach, then it shouldn't matter if they're pierceimmune or not.

Comment (ATH1909) [reply code/modules/mob/living/simple_animal/hostile/cockroach.dm:111]: you could add a special caltrop flag that makes it not affect pacifists.

**Comment by IndieanaJones** [comment]
More xenobio powercreep smh

**Comment by Mycah142** [comment]
Ant gang ant gang

File: code/modules/mob/living/simple_animal/hostile/cockroach.dm (hunk 2 above)
Comment (Ryll-Ryll) [reply code/modules/mob/living/simple_animal/hostile/cockroach.dm:111]: No

**Comment by kriskog** [comment]
Conflict

**Review by Twaticus — APPROVED** [review]
ok im laffin

**Pull request merged**